### PR TITLE
Allow external processing of invoices to be tracked

### PR DESCRIPTION
**Pull request by masqita**
- Adds a `processed_at` column
- Allows unprocessed invoices to be fetched
- Allows individual invoices to be marked as processed
- Update Billbo client for new feature

### CODE REVIEW

File: app/app.rb [hunk 1]
```diff
@@ -61,15 +63,32 @@ class App < Base
...
+  # e.g. email the invoice to the customer.
+  #
+  # number - The invoice number.
+  #
+  # Returns a JSON representation of the invoice
+  post '/invoices/:number/process' do
+    invoice = Invoice.where(number: params[:number]).first
+
+    halt 404 unless invoice
+    halt 400 if !invoice.pdf_generated? || invoice.processed?
```
Comment (challengee) [inline app/app.rb]: We probably shouldn't add the pdf generated constraint here if the unprocessed scope doesn't filter invoices on pdf generated. I would be inclined to remove this check and make processing a processed invoice a no-op. The client is then responsible for asking invoices that are pdf-generated + are unprocessed, making this more general.

We then probably also don't need the `ProcessingError`, as the client would be responsible for fetching only invoices that are processed as well. Keeps it simple.

Thoughts?

Comment (masqita) [reply app/app.rb]: Sure, I can follow your train of thought there. I envisioned more of a fixed progression of states, but leaving it open to the client works just as well and might allow for more use cases then those I had in mind.

**Comment by masqita** [comment]
@challengee made the necessary changes, should we bump the version to `1.1` too?

File: app/invoice.rb [hunk 2]
```diff
@@ -1,5 +1,6 @@
 class Invoice < Sequel::Model
   AlreadyFinalized = Class.new(StandardError)
+  ProcessingError = Class.new(StandardError)
```
Comment (challengee) [inline app/invoice.rb]: Do we still need this now? It's not used at the server side anymore. If you want to keep it we should probably move it to the client.

**Comment by challengee** [comment]
Feel free to merge if it looks good to you. We can bump the version, not that it has any meaning... We've not exactly been following any convention. This change is also non-breaking.

File: app/invoice.rb (hunk 2 above)
Comment (masqita) [reply app/invoice.rb]: Should have cleaned up house better, apologies…

**Pull request merged**